File: hitable_list.py
from hitable import Hitable

class HitableList(Hitable):
    def __init__(self, arr):
        self.arr = arr

    def hit(self, ray, tMin = 0.0, tMax = float("inf")):
        retRecord = None
        closest = tMax

        for item in self.arr:
            record = item.hit(ray, tMin, closest)
            if record:
                closest = record.t
                retRecord = record
        return retRecord

File: Sources/HitableList.swift
final class HitableList: Hitable {
  private let list: [Hitable]

  init(list: [Hitable]) {
    self.list = list
  }

  func hit(ray: Ray, tMin: Double = 0.0, tMax: Double = Double.infinity) -> HitRecord? {
    var retRecord: HitRecord?
    var closest = tMax

    list.forEach { object in
      if let record = object.hit(ray: ray, tMin: tMin, tMax: closest) {
        closest = record.t
        retRecord = record
      }
    }
    return retRecord
  }
}
